### DISCUSSION TITLE
Relative links instead of root-relative links for offline viewing

### DISCUSSION THREAD

**Discussion by kukimik** [srid/emanote#191] [Ideas]
I'd like to use `emanote` to create documents that are viewable offline (without starting a local web server; I just want to open the files in browser as `file://...`). The paths in the documents (e.g. hrefs coming from wiki links, the image src attributes) are relative to the root of the site, which is not suitable for this scenario. I've achieved my goal by running `python3 -m http.server` in the directory containing the html files generated by `emanote` and then downloading them using `wget --mirror --convert-links 'http://localhost:8000'`. But this is a workaround.

Is it difficult to make this customizable, so that the user could switch between relative and root-relative paths? I believe it may require some changes to `ema`.

**Comment by srid** [comment]
It is worth switching to relative links in HTML by default, I suppose. Are there any SEO implications?

For now, there is a workaround for your case: just create a new deployment layer, and specify `file:///home/kukimik/notesHtml` as the `baseUrl`. This approach is used in emanote-template; see 

https://github.com/srid/emanote-template/blob/554d0e2b8a905c68d40e9a1680033e5ba3d7dbaa/.deploy/github/index.yaml#L2

https://github.com/srid/emanote-template/blob/554d0e2b8a905c68d40e9a1680033e5ba3d7dbaa/.github/workflows/publish.yaml#L15

**Reply by kukimik** [reply]
> It is worth switching to relative links in HTML by default, I suppose. Are there any SEO implications?

Not sure. I don't have a SEO expert at hand, so maybe it'll need some googling.

> For now, there is a workaround for your case: 

Thanks! I'll try it out.
